stories/translator/doc-elements.tsx

```diff
@@ -54,10 +58,18 @@ export const DocElement = () => {
 
     const handleFormPreviewButton = () => {
         setPageMode("preview-mode");
+        if (formBuilderRef.current) {
+            const generatedSchema = formBuilderRef.current?.generateSchema();
+            setSchema(generatedSchema);
+        }
     };
 
     const handleSchemaViewButton = () => {
         setPageMode("schema-mode");
+        if (formBuilderRef.current) {
+            const generatedSchema = formBuilderRef.current?.generateSchema();
+            setSchema(generatedSchema);
+        }
     };
 
```

Comment:
I dont quite understand why the need to make these changes, what are we fixing with these changes?